application/src/main/java/org/opentripplanner/street/model/edge/StreetEdgeReluctanceCalculator.java

```diff
@@ -18,19 +18,18 @@ static double computeReluctance(
     boolean walkingBike,
     boolean edgeIsStairs
   ) {
-    if (edgeIsStairs) {
-      return pref.walk().stairsReluctance();
-    } else {
-      return switch (traverseMode) {
-        case WALK -> walkingBike ? pref.bike().walking().reluctance() : pref.walk().reluctance();
-        case BICYCLE -> pref.bike().reluctance();
-        case CAR -> pref.car().reluctance();
-        case SCOOTER -> pref.scooter().reluctance();
-        default -> throw new IllegalArgumentException(
-          "getReluctance(): Invalid mode " + traverseMode
-        );
-      };
-    }
+    return switch (traverseMode) {
+      case WALK -> walkingBike
+        ? pref.bike().walking().reluctance() *
+        (edgeIsStairs ? pref.bike().walking().stairsReluctance() : 1)
+        : pref.walk().reluctance() * (edgeIsStairs ? pref.walk().stairsReluctance() : 1);
```

Comment:
These nested ternary operations are a bit difficult to read. Maybe extract the logic into method(s).